drivers/led_strip/ws2812_uart.c

```
@@ -0,0 +1,336 @@
+
+	/* memset to 0xFF is correct for inverted signal logic */
+	memset(px_buf, 0xFF, buf_len);
+
+	/*
+	 * Convert pixel data into a packed bitstream for the UART.
+	 * Each color bit is expanded into a pattern of `bits_per_symbol`.
+	 */
+	for (size_t i = 0; i < num_pixels; i++) {
+		for (uint8_t j = 0; j < cfg->num_colors; j++) {
+			uint8_t pixel_val;
+
+			switch (cfg->color_mapping[j]) {
+			/* White channel is not supported by LED strip API. */
+			case LED_COLOR_ID_WHITE:
+				pixel_val = 0;
+				break;
+			case LED_COLOR_ID_RED:
+				pixel_val = pixels[i].r;
+				break;
+			case LED_COLOR_ID_GREEN:
+				pixel_val = pixels[i].g;
+				break;
+			case LED_COLOR_ID_BLUE:
+				pixel_val = pixels[i].b;
+				break;
+			default:
+				LOG_ERR("Invalid color mapping");
+				k_mutex_unlock(&data->lock);
+				return -EINVAL;
+			}
+
+			ws2812_uart_ser(pixel_val, cfg, &frame_bit_pos, &current_buf);
+		}
+	}
+
+	/*
+	 * Start the non-blocking transfer. The uart_tx function will return
+	 * immediately. The callback will signal completion via the semaphore.
+	 */
+	ret = uart_tx(cfg->uart_dev, px_buf, buf_len, SYS_FOREVER_US);
+	if (ret) {
+		k_mutex_unlock(&data->lock);
+		return ret;
+	}
+
+	/* Wait for the transfer to complete. */
+	k_sem_take(&data->tx_done_sem, K_FOREVER);
+
+	/* Latch the data and reset the strip */
+	ws2812_reset_delay(cfg->reset_delay);
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+
+static size_t ws2812_strip_length(const struct device *dev)
+{
+	const struct ws2812_uart_cfg *cfg = dev->config;
+
+	return cfg->length;
+}
+
+static int ws2812_uart_init(const struct device *dev)
+{
+	const struct ws2812_uart_cfg *cfg = dev->config;
+	struct ws2812_uart_data *data = dev->data;
+	int ret;
+
+	if (!device_is_ready(cfg->uart_dev)) {
+		LOG_ERR("%s: UART device %s not ready", dev->name, cfg->uart_dev->name);
+		return -ENODEV;
+	}
+
+	for (int i = 0; i < cfg->num_colors; i++) {
+		switch (cfg->color_mapping[i]) {
+		case LED_COLOR_ID_WHITE:
```

if not supported as above, then should return an error?